main.go

```
@@ -210,6 +212,14 @@ func (d *ElasticacheDiscovery) refresh(ctx context.Context) ([]*targetgroup.Grou
 		}
 
 		for _, cc := range o.CacheClusters {
+
+			if d.cfg.cacheClusterIDPattern != nil {
+				if stringMatch, _ := d.cfg.cacheClusterIDPattern.MatchString(*cc.CacheClusterId); !stringMatch {
```

Are we sure about ignoring the error here? Shouldn't we handle it?

---

Yep, this err is for the new regex library which catches issues with regex taking a long time to resolve.
Not a big issue for us, and we are not proposing to merge this to the official repo :p

---

FYI, this library was used because the go version does not support lookahead